src/doc/src/reference/semver.md

```diff
@@ -1078,6 +1079,73 @@ fn main() {
 }
 ```
 
+<a id="fn-unsafe-safe"></a>
+### Minor: making an `unsafe` function safe
+
+It is not a breaking change to make a previously `unsafe` function safe, as in
+the example below.
+
+Going the other way (making a safe function `unsafe`) is a breaking change.
+
+```rust,ignore
+// MINOR CHANGE
+
+///////////////////////////////////////////////////////////
+// Before
+pub unsafe fn foo() {}
+
+///////////////////////////////////////////////////////////
+// After
+pub fn foo() {}
+
+///////////////////////////////////////////////////////////
+// Example use of the library that will safely work.
+use updated_crate::foo;
+
+unsafe fn bar(f: unsafe fn()) {
+    f()
+}
+
+fn main() {
+    unsafe { foo() }; // The `unused_unsafe` lint will trigger here
+    unsafe { bar(foo) };
+}
+```
+
+Making an a previously `unsafe` associated function or method on structs /
+enums safe is also a minor change, while the same is not true for associated
+function on traits:
+
+```rust,ignore
+// MAJOR CHANGE
+
+///////////////////////////////////////////////////////////
+// Before
+pub trait Foo {
+    unsafe fn foo();
+}
+
+///////////////////////////////////////////////////////////
+// After
+pub trait Foo {
+    fn foo();
+}
+
+///////////////////////////////////////////////////////////
+// Example usage that will break.
+use updated_crate::Foo;
+
+struct Bar;
+
+impl Foo for Bar {
+    unsafe fn foo() {} // Error: method `foo` has an incompatible type for trait
+}
+```
+
+Note that local crates that have specified `#![deny(warnings)]` (which is an
+[anti-pattern][deny warnings]) will break, since they've explicitly opted out
+of Rust's stability guarantees.
```

Comment:
Can this avoid linking to the rust-unofficial site, and not take a stance on the use of `deny(warnings)`? That is a nuanced issue that I don't think the semver guidelines need to be explicit about. I think it's good to mention that this may cause the `unused_unsafe` lint to trigger (and link to it via `../../rustc/lints/listing/warn-by-default.html#unused-unsafe`), and that if that lint is denied, it may cause a compilation error.

We haven't made a decision yet on the behavior of lints. I think they should be allowed, but I'd like to discuss with the team more.

Reply:
You're probably right here